src/pybind/mgr/cephadm/service_discovery.py

```diff
@@ -276,8 +277,14 @@ def smb_sd_config(self) -> List[Dict[str, Collection[str]]]:
         srv_entries = []
         for dd in self.mgr.cache.get_daemons_by_type('smb'):
             assert dd.hostname is not None
+            try:
```

Comment:
Do we really need try-except block here? 
Since we're iterating over daemons  `self.mgr.cache.get_daemons_by_type('smb'): `, doesn't that imply the service and its spec should already exist?

Reply:
Fair point, but races haunt me and since this is a raw dict lookup (AFAIK) I'd rather fail gracefully here and fail to set up a service discovery endpoint than debug a race that throws a hard to track down traceback occasionally.

Reply:
Makes sense! In that case, maybe we could add a warning log that the spec was unexpectedly missing?